server.js

```diff
@@ -34,29 +36,41 @@ const userMessageTimes = {};
 
 /* Handle incoming messages */
 bot.on("message", msg => {
-    const query = msg.content.substr(commandChar.length).split(" ");
-    const command = query[0].toLowerCase();
-    const parameter = query.length > 1 ? query.slice(1).join(" ") : "";
+    // generate RegExp pattern for message parsing
+    // Example: ^!(card|price) ?(.*)$|!(card|price) ?([^!]*)(!|$)
+    const charPattern = _.escapeRegExp(commandChar);
+    const commandPattern = charPattern+'('+Object.keys(handlers).map(_.escapeRegExp).join('|')+')';
+    const regExpPattern = `^${commandPattern} ?(.*)$|${commandPattern} ?([^${charPattern}]*)(${charPattern}|$)`;
+
+    const queries = msg.content.match(new RegExp(regExpPattern, 'ig')) || [];
```

Comment:
Moved the RegExp generation to a separate function.
We can't use capture groups because JS doesn't support capture groups together with the `g` flag for RegExp `.match`, hence the manual splitting.